pytest/test_algo.py

```diff
@@ -117,4 +117,81 @@ def test_task331a_main_logic(number, expected_value):
     (75, ['1^2 + 5^2 + 7^2', '1^2 + 7^2 + 5^2', '5^2 + 1^2 + 7^2', '5^2 + 5^2 + 5^2', '5^2 + 7^2 + 1^2', '7^2 + 1^2 + 5^2', '7^2 + 5^2 + 1^2']),
     (4, False)])
 def test_task331b_main_logic(number, expected_value):
-    assert algo.Task331b.main_logic(number) == expected_value
\ No newline at end of file
+    assert algo.Task331b.main_logic(number) == expected_value
+
+
+# Testing base class with validation method for exceptions
+@pytest.mark.parametrize('data, expected_value', [('1.2', ValueError), ('-3ssss', ValueError), ('0', ValueError),
+                                                  ('445', ValueError), ('7m 456', TypeError), ('7115m 4', TypeError),
+                                                  ('124 4d56', TypeError), ('7,5555 456', TypeError),
```

Comment:
Here I see this trouble too, equal lines would me more readable

Reply:
ok